In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
%cd drive/MyDrive/pyFM

/content/drive/MyDrive/pyFM


In [ ]:
!pip install potpourri3d robust_laplacian open3d trimesh

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 858 kB 14.7 MB/s 
     |████████████████████████████████| 388 kB 69.0 MB/s 
     |████████████████████████████████| 408.6 MB 32 kB/s 
     |████████████████████████████████| 646 kB 59.6 MB/s 
     |████████████████████████████████| 8.8 MB 53.0 MB/s 
     |████████████████████████████████| 3.1 MB 42.6 MB/s 
     |████████████████████████████████| 596 kB 57.6 MB/s 
     |████████████████████████████████| 1.1 MB 46.6 MB/s 
     |████████████████████████████████| 428 kB 54.6 MB/s 
     |████████████████████████████████| 54 kB 1.6 MB/s 
     |████████████████████████████████| 344 kB 59.1 MB/s 
     |████████████████████████████████| 1.2 MB 57.1 MB/s 
     |████████████████████████████████| 561 kB 71.4 MB/s 
     |████████████████████████████████| 54 kB 3.5 MB/s 
     |████████████████████████████████| 131 kB 75.5 MB/s 
     |█████████████████████████████

In [ ]:
import yaml

from project.datasets import Tosca
from project.train_dpfm import train_net
from project.eval_dpfm import eval_net

In [ ]:
cfg = yaml.safe_load(open(f"dpfm/config/tosca_cuts.yaml", "r"))

In [ ]:
train_net(cfg)

using dataset cache path: data/SGP_dataset_off/train/cache_cuts_train.pt
  --> loading dataset from cache
start training


  0%|          | 1/300 [00:16<1:22:35, 16.57s/it]

#epoch:2, #batch:24, #iteration:50, loss:302.1380615234375


  1%|          | 3/300 [00:44<1:11:43, 14.49s/it]

#epoch:4, #batch:22, #iteration:100, loss:137.3421630859375


  2%|▏         | 5/300 [01:12<1:09:29, 14.13s/it]

#epoch:6, #batch:20, #iteration:150, loss:227.33448791503906


  2%|▏         | 7/300 [01:38<1:07:00, 13.72s/it]

#epoch:8, #batch:18, #iteration:200, loss:164.47032165527344


  3%|▎         | 9/300 [02:05<1:05:19, 13.47s/it]

#epoch:10, #batch:16, #iteration:250, loss:127.76302337646484


  4%|▎         | 11/300 [02:31<1:04:25, 13.37s/it]

#epoch:12, #batch:14, #iteration:300, loss:136.3143310546875


  4%|▍         | 13/300 [02:58<1:03:45, 13.33s/it]

#epoch:14, #batch:12, #iteration:350, loss:168.7238311767578


  5%|▌         | 15/300 [03:25<1:03:09, 13.30s/it]

#epoch:16, #batch:10, #iteration:400, loss:142.653564453125


  6%|▌         | 17/300 [03:51<1:02:34, 13.27s/it]

#epoch:18, #batch:8, #iteration:450, loss:125.97071838378906


  6%|▋         | 19/300 [04:17<1:02:05, 13.26s/it]

#epoch:20, #batch:6, #iteration:500, loss:181.5066680908203


  7%|▋         | 21/300 [04:44<1:01:46, 13.28s/it]

#epoch:22, #batch:4, #iteration:550, loss:48.7037467956543


  8%|▊         | 23/300 [05:11<1:01:15, 13.27s/it]

#epoch:24, #batch:2, #iteration:600, loss:116.51419830322266


  8%|▊         | 25/300 [05:37<1:00:56, 13.29s/it]

#epoch:25, #batch:26, #iteration:650, loss:27.822362899780273


  9%|▊         | 26/300 [05:50<1:00:33, 13.26s/it]

#epoch:27, #batch:24, #iteration:700, loss:93.96780395507812


  9%|▉         | 28/300 [06:17<59:53, 13.21s/it]  

#epoch:29, #batch:22, #iteration:750, loss:61.507080078125


 10%|█         | 30/300 [06:44<1:00:23, 13.42s/it]

#epoch:31, #batch:20, #iteration:800, loss:121.80431365966797


 11%|█         | 32/300 [07:10<59:20, 13.29s/it]

#epoch:33, #batch:18, #iteration:850, loss:41.24979782104492


 11%|█▏        | 34/300 [07:36<58:27, 13.18s/it]

#epoch:35, #batch:16, #iteration:900, loss:83.79447937011719


 12%|█▏        | 36/300 [08:02<57:39, 13.10s/it]

#epoch:37, #batch:14, #iteration:950, loss:44.960113525390625


 13%|█▎        | 38/300 [08:29<57:06, 13.08s/it]

#epoch:39, #batch:12, #iteration:1000, loss:35.95684814453125


 13%|█▎        | 40/300 [08:55<56:37, 13.07s/it]

#epoch:41, #batch:10, #iteration:1050, loss:63.46916961669922


 14%|█▍        | 42/300 [09:21<55:59, 13.02s/it]

#epoch:43, #batch:8, #iteration:1100, loss:101.66444396972656


 15%|█▍        | 44/300 [09:47<55:45, 13.07s/it]

#epoch:45, #batch:6, #iteration:1150, loss:49.94754409790039


 15%|█▌        | 46/300 [10:13<55:16, 13.06s/it]

#epoch:47, #batch:4, #iteration:1200, loss:27.584152221679688


 16%|█▌        | 48/300 [10:39<54:42, 13.03s/it]

#epoch:49, #batch:2, #iteration:1250, loss:89.13593292236328


 17%|█▋        | 50/300 [11:05<54:07, 12.99s/it]

#epoch:50, #batch:26, #iteration:1300, loss:77.7473373413086


 17%|█▋        | 51/300 [11:18<53:50, 12.98s/it]

#epoch:52, #batch:24, #iteration:1350, loss:59.160404205322266


 18%|█▊        | 53/300 [11:45<54:10, 13.16s/it]

#epoch:54, #batch:22, #iteration:1400, loss:64.67733764648438


 18%|█▊        | 55/300 [12:11<53:28, 13.10s/it]

#epoch:56, #batch:20, #iteration:1450, loss:88.97178649902344


 19%|█▉        | 57/300 [12:37<52:42, 13.01s/it]

#epoch:58, #batch:18, #iteration:1500, loss:31.59101104736328


 20%|█▉        | 59/300 [13:03<52:26, 13.06s/it]

#epoch:60, #batch:16, #iteration:1550, loss:42.48494338989258


 20%|██        | 61/300 [13:29<51:52, 13.02s/it]

#epoch:62, #batch:14, #iteration:1600, loss:24.936904907226562


 21%|██        | 63/300 [13:55<51:29, 13.04s/it]

#epoch:64, #batch:12, #iteration:1650, loss:20.657472610473633


 22%|██▏       | 65/300 [14:21<50:52, 12.99s/it]

#epoch:66, #batch:10, #iteration:1700, loss:35.09893035888672


 22%|██▏       | 67/300 [14:47<50:42, 13.06s/it]

#epoch:68, #batch:8, #iteration:1750, loss:16.902875900268555


 23%|██▎       | 69/300 [15:13<50:08, 13.02s/it]

#epoch:70, #batch:6, #iteration:1800, loss:12.676803588867188


 24%|██▎       | 71/300 [15:39<49:32, 12.98s/it]

#epoch:72, #batch:4, #iteration:1850, loss:15.724146842956543


 24%|██▍       | 73/300 [16:05<49:11, 13.00s/it]

#epoch:74, #batch:2, #iteration:1900, loss:34.69768524169922


 25%|██▌       | 75/300 [16:31<48:48, 13.02s/it]

#epoch:75, #batch:26, #iteration:1950, loss:59.663909912109375


 25%|██▌       | 76/300 [16:44<48:39, 13.03s/it]

#epoch:77, #batch:24, #iteration:2000, loss:21.312219619750977


 26%|██▌       | 78/300 [17:11<48:59, 13.24s/it]

#epoch:79, #batch:22, #iteration:2050, loss:12.305496215820312


 27%|██▋       | 80/300 [17:37<48:14, 13.16s/it]

#epoch:81, #batch:20, #iteration:2100, loss:15.828622817993164


 27%|██▋       | 82/300 [18:03<47:42, 13.13s/it]

#epoch:83, #batch:18, #iteration:2150, loss:12.268756866455078


 28%|██▊       | 84/300 [18:29<47:18, 13.14s/it]

#epoch:85, #batch:16, #iteration:2200, loss:30.555055618286133


 29%|██▊       | 86/300 [18:56<46:47, 13.12s/it]

#epoch:87, #batch:14, #iteration:2250, loss:28.11054039001465


 29%|██▉       | 88/300 [19:22<46:22, 13.13s/it]

#epoch:89, #batch:12, #iteration:2300, loss:7.948394775390625


 30%|███       | 90/300 [19:48<46:01, 13.15s/it]

#epoch:91, #batch:10, #iteration:2350, loss:15.584312438964844


 31%|███       | 92/300 [20:15<45:35, 13.15s/it]

#epoch:93, #batch:8, #iteration:2400, loss:38.19013595581055


 31%|███▏      | 94/300 [20:41<44:59, 13.10s/it]

#epoch:95, #batch:6, #iteration:2450, loss:30.541929244995117


 32%|███▏      | 96/300 [21:07<44:35, 13.12s/it]

#epoch:97, #batch:4, #iteration:2500, loss:19.49611473083496


 33%|███▎      | 98/300 [21:33<44:04, 13.09s/it]

#epoch:99, #batch:2, #iteration:2550, loss:27.645299911499023


 33%|███▎      | 99/300 [21:46<43:54, 13.10s/it]

Decaying learning rate, new one: 0.0005


 33%|███▎      | 100/300 [21:59<43:41, 13.11s/it]

#epoch:100, #batch:26, #iteration:2600, loss:15.579360961914062


 34%|███▎      | 101/300 [22:13<43:31, 13.13s/it]

#epoch:102, #batch:24, #iteration:2650, loss:25.00315284729004


 34%|███▍      | 103/300 [22:39<43:37, 13.28s/it]

#epoch:104, #batch:22, #iteration:2700, loss:14.389491081237793


 35%|███▌      | 105/300 [23:06<42:49, 13.17s/it]

#epoch:106, #batch:20, #iteration:2750, loss:29.144506454467773


 36%|███▌      | 107/300 [23:32<42:06, 13.09s/it]

#epoch:108, #batch:18, #iteration:2800, loss:24.462148666381836


 36%|███▋      | 109/300 [23:58<41:39, 13.09s/it]

#epoch:110, #batch:16, #iteration:2850, loss:14.317612648010254


 37%|███▋      | 111/300 [24:24<41:09, 13.07s/it]

#epoch:112, #batch:14, #iteration:2900, loss:25.653287887573242


 38%|███▊      | 113/300 [24:50<40:44, 13.07s/it]

#epoch:114, #batch:12, #iteration:2950, loss:28.276206970214844


 38%|███▊      | 115/300 [25:16<40:27, 13.12s/it]

#epoch:116, #batch:10, #iteration:3000, loss:25.086795806884766


 39%|███▉      | 117/300 [25:42<39:50, 13.06s/it]

#epoch:118, #batch:8, #iteration:3050, loss:14.475030899047852


 40%|███▉      | 119/300 [26:08<39:16, 13.02s/it]

#epoch:120, #batch:6, #iteration:3100, loss:13.097270011901855


 40%|████      | 121/300 [26:34<38:49, 13.02s/it]

#epoch:122, #batch:4, #iteration:3150, loss:24.535221099853516


 41%|████      | 123/300 [27:01<38:26, 13.03s/it]

#epoch:124, #batch:2, #iteration:3200, loss:14.243640899658203


 42%|████▏     | 125/300 [27:27<37:58, 13.02s/it]

#epoch:125, #batch:26, #iteration:3250, loss:15.585161209106445


 42%|████▏     | 126/300 [27:40<37:57, 13.09s/it]

#epoch:127, #batch:24, #iteration:3300, loss:25.452503204345703


 43%|████▎     | 128/300 [28:06<37:31, 13.09s/it]

#epoch:129, #batch:22, #iteration:3350, loss:24.343549728393555


 43%|████▎     | 130/300 [28:32<36:56, 13.04s/it]

#epoch:131, #batch:20, #iteration:3400, loss:24.518686294555664


 44%|████▍     | 132/300 [28:58<36:21, 12.99s/it]

#epoch:133, #batch:18, #iteration:3450, loss:24.627452850341797


 45%|████▍     | 134/300 [29:24<35:57, 13.00s/it]

#epoch:135, #batch:16, #iteration:3500, loss:14.335460662841797


 45%|████▌     | 136/300 [29:50<35:25, 12.96s/it]

#epoch:137, #batch:14, #iteration:3550, loss:25.137208938598633


 46%|████▌     | 138/300 [30:16<35:09, 13.02s/it]

#epoch:139, #batch:12, #iteration:3600, loss:25.03306770324707


 47%|████▋     | 140/300 [30:42<34:32, 12.95s/it]

#epoch:141, #batch:10, #iteration:3650, loss:19.093097686767578


 47%|████▋     | 142/300 [31:08<34:07, 12.96s/it]

#epoch:143, #batch:8, #iteration:3700, loss:19.091590881347656


 48%|████▊     | 144/300 [31:34<33:40, 12.95s/it]

#epoch:145, #batch:6, #iteration:3750, loss:27.777515411376953


 49%|████▊     | 146/300 [32:00<33:19, 12.99s/it]

#epoch:147, #batch:4, #iteration:3800, loss:28.781558990478516


 49%|████▉     | 148/300 [32:25<32:49, 12.95s/it]

#epoch:149, #batch:2, #iteration:3850, loss:28.074480056762695


 50%|█████     | 150/300 [32:52<32:48, 13.12s/it]

#epoch:150, #batch:26, #iteration:3900, loss:19.152935028076172


 50%|█████     | 151/300 [33:05<32:26, 13.06s/it]

#epoch:152, #batch:24, #iteration:3950, loss:25.075313568115234


 51%|█████     | 153/300 [33:31<31:50, 13.00s/it]

#epoch:154, #batch:22, #iteration:4000, loss:35.166988372802734


 52%|█████▏    | 155/300 [33:57<31:23, 12.99s/it]

#epoch:156, #batch:20, #iteration:4050, loss:6.5308837890625


 52%|█████▏    | 157/300 [34:23<30:52, 12.95s/it]

#epoch:158, #batch:18, #iteration:4100, loss:25.062089920043945


 53%|█████▎    | 159/300 [34:49<30:28, 12.97s/it]

#epoch:160, #batch:16, #iteration:4150, loss:51.89509582519531


 54%|█████▎    | 161/300 [35:14<29:58, 12.94s/it]

#epoch:162, #batch:14, #iteration:4200, loss:24.797286987304688


 54%|█████▍    | 163/300 [35:40<29:33, 12.94s/it]

#epoch:164, #batch:12, #iteration:4250, loss:24.53350067138672


 55%|█████▌    | 165/300 [36:07<29:17, 13.02s/it]

#epoch:166, #batch:10, #iteration:4300, loss:12.879486083984375


 56%|█████▌    | 167/300 [36:33<28:51, 13.02s/it]

#epoch:168, #batch:8, #iteration:4350, loss:27.870990753173828


 56%|█████▋    | 169/300 [37:00<29:08, 13.35s/it]

#epoch:170, #batch:6, #iteration:4400, loss:14.632176399230957


 57%|█████▋    | 171/300 [37:27<28:53, 13.44s/it]

#epoch:172, #batch:4, #iteration:4450, loss:26.457983016967773


 58%|█████▊    | 173/300 [37:53<28:03, 13.25s/it]

#epoch:174, #batch:2, #iteration:4500, loss:26.00847816467285


 58%|█████▊    | 175/300 [38:19<27:25, 13.16s/it]

#epoch:175, #batch:26, #iteration:4550, loss:7.138396739959717


 59%|█████▊    | 176/300 [38:33<27:38, 13.37s/it]

#epoch:177, #batch:24, #iteration:4600, loss:14.282597541809082


 59%|█████▉    | 178/300 [38:59<26:53, 13.22s/it]

#epoch:179, #batch:22, #iteration:4650, loss:25.272008895874023


 60%|██████    | 180/300 [39:26<26:24, 13.21s/it]

#epoch:181, #batch:20, #iteration:4700, loss:17.438119888305664


 61%|██████    | 182/300 [39:52<25:53, 13.17s/it]

#epoch:183, #batch:18, #iteration:4750, loss:28.220428466796875


 61%|██████▏   | 184/300 [40:18<25:27, 13.17s/it]

#epoch:185, #batch:16, #iteration:4800, loss:16.56249237060547


 62%|██████▏   | 186/300 [40:45<25:02, 13.18s/it]

#epoch:187, #batch:14, #iteration:4850, loss:25.814666748046875


 63%|██████▎   | 188/300 [41:11<24:31, 13.14s/it]

#epoch:189, #batch:12, #iteration:4900, loss:26.17914581298828


 63%|██████▎   | 190/300 [41:37<24:05, 13.14s/it]

#epoch:191, #batch:10, #iteration:4950, loss:15.601170539855957


 64%|██████▍   | 192/300 [42:04<23:40, 13.16s/it]

#epoch:193, #batch:8, #iteration:5000, loss:15.588129043579102


 65%|██████▍   | 194/300 [42:30<23:15, 13.17s/it]

#epoch:195, #batch:6, #iteration:5050, loss:15.586482048034668


 65%|██████▌   | 196/300 [42:56<22:49, 13.17s/it]

#epoch:197, #batch:4, #iteration:5100, loss:24.39560890197754


 66%|██████▌   | 198/300 [43:25<23:23, 13.76s/it]

#epoch:199, #batch:2, #iteration:5150, loss:19.902589797973633


 66%|██████▋   | 199/300 [43:38<22:52, 13.59s/it]

Decaying learning rate, new one: 0.00025


 67%|██████▋   | 200/300 [43:52<22:48, 13.69s/it]

#epoch:200, #batch:26, #iteration:5200, loss:15.587831497192383


 67%|██████▋   | 201/300 [44:05<22:28, 13.62s/it]

#epoch:202, #batch:24, #iteration:5250, loss:14.775321960449219


 68%|██████▊   | 203/300 [44:32<21:40, 13.40s/it]

#epoch:204, #batch:22, #iteration:5300, loss:25.390830993652344


 68%|██████▊   | 205/300 [44:58<21:02, 13.29s/it]

#epoch:206, #batch:20, #iteration:5350, loss:33.438331604003906


 69%|██████▉   | 207/300 [45:24<20:31, 13.24s/it]

#epoch:208, #batch:18, #iteration:5400, loss:24.53113555908203


 70%|██████▉   | 209/300 [45:50<19:58, 13.17s/it]

#epoch:210, #batch:16, #iteration:5450, loss:34.611419677734375


 70%|███████   | 211/300 [46:17<19:24, 13.08s/it]

#epoch:212, #batch:14, #iteration:5500, loss:24.535541534423828


 71%|███████   | 213/300 [46:42<18:53, 13.03s/it]

#epoch:214, #batch:12, #iteration:5550, loss:16.576292037963867


 72%|███████▏  | 215/300 [47:09<18:27, 13.04s/it]

#epoch:216, #batch:10, #iteration:5600, loss:26.257495880126953


 72%|███████▏  | 217/300 [47:35<18:05, 13.08s/it]

#epoch:218, #batch:8, #iteration:5650, loss:25.410110473632812


 73%|███████▎  | 219/300 [48:01<17:39, 13.08s/it]

#epoch:220, #batch:6, #iteration:5700, loss:25.129642486572266


 74%|███████▎  | 221/300 [48:27<17:13, 13.08s/it]

#epoch:222, #batch:4, #iteration:5750, loss:14.636011123657227


 74%|███████▍  | 223/300 [48:53<16:47, 13.09s/it]

#epoch:224, #batch:2, #iteration:5800, loss:27.83881378173828


 75%|███████▌  | 225/300 [49:20<16:26, 13.16s/it]

#epoch:225, #batch:26, #iteration:5850, loss:19.904443740844727


 75%|███████▌  | 226/300 [49:33<16:26, 13.33s/it]

#epoch:227, #batch:24, #iteration:5900, loss:34.77013397216797


 76%|███████▌  | 228/300 [50:00<15:53, 13.25s/it]

#epoch:229, #batch:22, #iteration:5950, loss:24.83745765686035


 77%|███████▋  | 230/300 [50:26<15:19, 13.13s/it]

#epoch:231, #batch:20, #iteration:6000, loss:28.071401596069336


 77%|███████▋  | 232/300 [50:52<14:54, 13.15s/it]

#epoch:233, #batch:18, #iteration:6050, loss:14.78931999206543


 78%|███████▊  | 234/300 [51:18<14:23, 13.09s/it]

#epoch:235, #batch:16, #iteration:6100, loss:14.338919639587402


 79%|███████▊  | 236/300 [51:44<13:51, 13.00s/it]

#epoch:237, #batch:14, #iteration:6150, loss:13.307071685791016


 79%|███████▉  | 238/300 [52:10<13:27, 13.03s/it]

#epoch:239, #batch:12, #iteration:6200, loss:25.52759552001953


 80%|████████  | 240/300 [52:36<13:04, 13.07s/it]

#epoch:241, #batch:10, #iteration:6250, loss:19.136423110961914


 81%|████████  | 242/300 [53:02<12:38, 13.08s/it]

#epoch:243, #batch:8, #iteration:6300, loss:14.300019264221191


 81%|████████▏ | 244/300 [53:29<12:12, 13.08s/it]

#epoch:245, #batch:6, #iteration:6350, loss:24.96074104309082


 82%|████████▏ | 246/300 [53:55<11:45, 13.07s/it]

#epoch:247, #batch:4, #iteration:6400, loss:14.333869934082031


 83%|████████▎ | 248/300 [54:22<11:29, 13.27s/it]

#epoch:249, #batch:2, #iteration:6450, loss:25.076278686523438


 83%|████████▎ | 250/300 [54:48<11:01, 13.23s/it]

#epoch:250, #batch:26, #iteration:6500, loss:12.23309326171875


 84%|████████▎ | 251/300 [55:01<10:46, 13.19s/it]

#epoch:252, #batch:24, #iteration:6550, loss:25.49689292907715


 84%|████████▍ | 253/300 [55:27<10:16, 13.12s/it]

#epoch:254, #batch:22, #iteration:6600, loss:12.87265396118164


 85%|████████▌ | 255/300 [55:53<09:48, 13.08s/it]

#epoch:256, #batch:20, #iteration:6650, loss:19.32501983642578


 86%|████████▌ | 257/300 [56:20<09:22, 13.08s/it]

#epoch:258, #batch:18, #iteration:6700, loss:35.043365478515625


 86%|████████▋ | 259/300 [56:46<08:56, 13.08s/it]

#epoch:260, #batch:16, #iteration:6750, loss:6.617563724517822


 87%|████████▋ | 261/300 [57:12<08:30, 13.10s/it]

#epoch:262, #batch:14, #iteration:6800, loss:12.869109153747559


 88%|████████▊ | 263/300 [57:38<08:05, 13.12s/it]

#epoch:264, #batch:12, #iteration:6850, loss:19.90274429321289


 88%|████████▊ | 265/300 [58:04<07:39, 13.12s/it]

#epoch:266, #batch:10, #iteration:6900, loss:14.29836368560791


 89%|████████▉ | 267/300 [58:31<07:14, 13.15s/it]

#epoch:268, #batch:8, #iteration:6950, loss:24.43140983581543


 90%|████████▉ | 269/300 [58:57<06:48, 13.19s/it]

#epoch:270, #batch:6, #iteration:7000, loss:24.35105323791504


 90%|█████████ | 271/300 [59:23<06:21, 13.14s/it]

#epoch:272, #batch:4, #iteration:7050, loss:19.104633331298828


 91%|█████████ | 273/300 [59:50<06:00, 13.35s/it]

#epoch:274, #batch:2, #iteration:7100, loss:7.142032623291016


 92%|█████████▏| 275/300 [1:00:17<05:30, 13.22s/it]

#epoch:275, #batch:26, #iteration:7150, loss:24.817401885986328


 92%|█████████▏| 276/300 [1:00:30<05:16, 13.19s/it]

#epoch:277, #batch:24, #iteration:7200, loss:25.07118797302246


 93%|█████████▎| 278/300 [1:00:56<04:48, 13.12s/it]

#epoch:279, #batch:22, #iteration:7250, loss:26.460420608520508


 93%|█████████▎| 280/300 [1:01:22<04:22, 13.14s/it]

#epoch:281, #batch:20, #iteration:7300, loss:25.66828155517578


 94%|█████████▍| 282/300 [1:01:48<03:55, 13.10s/it]

#epoch:283, #batch:18, #iteration:7350, loss:14.950443267822266


 95%|█████████▍| 284/300 [1:02:14<03:29, 13.09s/it]

#epoch:285, #batch:16, #iteration:7400, loss:12.87553882598877


 95%|█████████▌| 286/300 [1:02:41<03:03, 13.10s/it]

#epoch:287, #batch:14, #iteration:7450, loss:27.84998321533203


 96%|█████████▌| 288/300 [1:03:07<02:37, 13.09s/it]

#epoch:289, #batch:12, #iteration:7500, loss:24.35024642944336


 97%|█████████▋| 290/300 [1:03:33<02:11, 13.11s/it]

#epoch:291, #batch:10, #iteration:7550, loss:6.5267333984375


 97%|█████████▋| 292/300 [1:03:59<01:44, 13.10s/it]

#epoch:293, #batch:8, #iteration:7600, loss:25.05594825744629


 98%|█████████▊| 294/300 [1:04:25<01:18, 13.10s/it]

#epoch:295, #batch:6, #iteration:7650, loss:24.81686782836914


 99%|█████████▊| 296/300 [1:04:52<00:53, 13.30s/it]

#epoch:297, #batch:4, #iteration:7700, loss:23.641454696655273


 99%|█████████▉| 298/300 [1:05:18<00:26, 13.17s/it]

#epoch:299, #batch:2, #iteration:7750, loss:19.89952850341797


100%|█████████▉| 299/300 [1:05:31<00:13, 13.16s/it]

Decaying learning rate, new one: 0.000125


100%|██████████| 300/300 [1:05:44<00:00, 13.15s/it]

#epoch:300, #batch:26, #iteration:7800, loss:12.239409446716309


In [ ]:
eval_net(cfg, f"project/data/saved_models_cuts_2/ep" + f"_{279}.pth", "data/eval/first_test.pt")

using dataset cache path: data/SGP_dataset_off/test/cache_cuts_train.pt
  --> loading dataset from cache


100%|██████████| 6/6 [03:46<00:00, 37.73s/it]

Mean normalized geodesic error: 0.6637012437052892


# Notes
- Trained on 26 shapes, tested on 6 shapes (80/20 train/test split)
- Training process seems to work well, can't tell if it's overfitting due to lack of validation implementation
- Compared to FM with hand-crafted descriptors (WKS): 5.99 <-> 0.66 (mean normalized geodesic error)
- Improvements:
  - validation set implementation
  - tensorboard visualization
  - lr scheduler from pytorch
  - inference is slow, mostly done on CPU, transfer to GPU
- Next steps:
  - double check all dataloaders & FM to p2p conversions
  - training on whole partial-to-full dataset (TOSCA), or mix full-to-full & partial-to-full, e.g. warm start with model pre-trained on an easier problem, i.e. full-to-full
  - combine model w/ coupled FM
